components/solr-access-service/api/src/main/java/org/phenotips/solr/HPOScriptService.java

```diff
@@ -78,12 +78,16 @@
             crt = nodes.poll();
             results.add(String.valueOf(crt.get(ID_FIELD_NAME)));
             @SuppressWarnings("unchecked")
-            List<String> parents = (List<String>) crt.get("is_a");
+            Object parents = crt.get("is_a");
             if (parents == null) {
                 continue;
             }
-            for (String pid : parents) {
-                nodes.add(this.get(StringUtils.substringBefore(pid, " ")));
+            if (parents.getClass() == String.class) {
```

Comment:
Actually, what you should do is check if the object is a String, and if it is, you should turn it into a list using [`Collections.singletonList()`](http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#singletonList%28T%29). That way, you can keep the for loop, and don't have to duplicate the actual node adding logic.